### ISSUE TITLE
SSL support

### ISSUE THREAD

**Issue by f213** [datanymizer/datanymizer#61]
I am trying to use datanymizer in the environment that requires SSL connection (Heroku postgres). It fails with an error:

```
$ docker run --rm -v /Users/f213/prj/education/dev-db:/app -w /app datanymizer/pg_datanymizer "postgres://user:pwd@<REDACTED>.eu-west-1.compute.amazonaws.com:5432/<REDACTED>?sslmode=require"
Error: error performing TLS handshake: no TLS implementation configured

Caused by:
    no TLS implementation configured
```

Any chances this great project will support SSL?

**Comment by evgeniy-r** [comment]
Hello!

I think we will add this option (unless there are any unexpected troubles).

**Comment by evgeniy-r** [comment]
Hello again!

I add the basic support of SSL (in the `main` branch). I have tested it with `sslmode=require`.
Let us know if something will be wrong, please.

**Comment by f213** [comment]
That's great! It would be nice to add an option for disabling server certificate validation, so i don't have to put my CA in the dockerfile

**Comment by evgeniy-r** [comment]
I thought it is controlled by this option: https://www.postgresql.org/docs/13/libpq-ssl.html#LIBPQ-SSL-SSLMODE-STATEMENTS,
am I wrong?

**Comment by f213** [comment]
When i try to set sslmode to `allow`. i get
```
Error: invalid connection string: invalid value for option `sslmode`

Caused by:
    invalid value for option `sslmode`
```

What i set it to `prefer`, i get this:
```
Error: error performing TLS handshake: error:1416F086:SSL routines:tls_process_server_certificate:certificate verify failed:../ssl/statem/statem_clnt.c:1915: (self signed certificate)

Caused by:
    0: error:1416F086:SSL routines:tls_process_server_certificate:certificate verify failed:../ssl/statem/statem_clnt.c:1915: (self signed certificate)
    1: error:1416F086:SSL routines:tls_process_server_certificate:certificate verify failed:../ssl/statem/statem_clnt.c:1915:
```

**Comment by evgeniy-r** [comment]
OK, I will take a look.

**Comment by evgeniy-r** [comment]
The client we are using does not support all SSL modes (https://docs.rs/tokio-postgres/0.7.2/tokio_postgres/config/enum.SslMode.html). This is the reason why `allow`does not work. 

I have add the new CLI options here - https://github.com/datanymizer/datanymizer/pull/64/ 
* `--accept_invalid_hostnames`
* `--accept_invalid_certs`

I tested this options, can you check them too?

`pg_datanimizer ..... --accept_invalid_certs=true`

**Comment by f213** [comment]
It works like a charm with my Heroku postgres instance, thank you so much!

**Comment by evgeniy-r** [comment]
The PR #64 was merged in the main branch.

**Comment by marmik18** [comment]
When will these changes be released? 
When I try to use the flag `--accept_invalid_certs=true` it throws this error so I am assuming the changes haven't been released yet. 
```
error: Found argument '--accept_invalid_certs' which wasn't expected, or isn't valid in this context
```

**Comment by evgeniy-r** [comment]
Hello!

This feature is not in the release yet.
We plan to make a new release in a few days (the beginning of the next week).

For now you can use a version from the `main` branch.

**Comment by marmik18** [comment]
I am planning to run this on a CI/CD pipeline.
how can I build it from the `main` branch in the pipeline?
Sorry to ask a naive question, but I don't have experience in rust.

**Comment by evgeniy-r** [comment]
1. You can build a docker image (with `docker build`), push it to some docker registry and then use in your pipeline.
The project contains `Dockerfile`.

2. You can build the binary. You should install [Rust](https://www.rust-lang.org/tools/install) (via `rustup`).
Then clone the repo and build the binary:
```shell
git checkout git@github.com:datanymizer/datanymizer.git
cd datanymizer
cargo build --release
```
The `pg_datanymizer` binary will be in the `target/release` subdirectory.
You can use it in the pipeline.
If you need the binary for some platform other then yours (for example, `musl`):
```shell
rustup target add x86_64-unknown-linux-musl
cargo build --release --target x86_64-unknown-linux-musl
```
[Here](https://doc.rust-lang.org/nightly/rustc/platform-support.html) is a list of platforms.

3. Just wait a day or two.

**Comment by marmik18** [comment]
Thank you so much for the detailed answer. 
For now, I will use the `Dockerfile` to build the image and then update to the released version when you release the update.

**Comment by evgeniy-r** [comment]
We released the new version (0.3.1).

**Comment by marmik18** [comment]
@evgeniy-r thank you for informing, I will test it out today.
